src/databases/paraDIS/RC_cpp_lib/pathutil.h

```diff
@@ -14,7 +18,14 @@
   Dirname("/") == "/"
 
 */ 
-inline string Dirname(string filename) {
+inline string Dirname(string filename)
+{
+#ifdef RC_CPP_VISIT_BUILD
```

Comment:
Does this new code block **NOT** work on any platform you are aware of? If not, then I'd opt to get rid of anything related to the *old* way of building the code including any conditional compilation blocks like this. In fact, I couldn't tell for sure but it looks like there are a number of source files in `RC_cpp_lib` dir that have been removed from consideration for build purposes. Will it even build the *old* way now? Will we care? I don't think so. Just make the new way of building the only way it gets built. My two cents.

Reply:
The old code block should work on linux. It just didn't compile on Windows. Since we have the same functionality, I opted to use it. I could have made the  ifdef  for Windows specifically, but chose not.  

AFAIK, all the files in the subdirs of this plugin are directly from the paradis lib source code.  

Yes, there are now source files in these subdirs that aren't part of compilation for the VisIt plugin. They didn't seem to be necessary, and I didn't want to take the time to port all the source to Windows.

Reply:
Right. So why not get rid of the old code block entire?

Reply:
My thought process regarding not getting rid of old code ... since this is (I think) part of the original paradis library source, if it ever needs to be compared to newer paradis for updates to the reader, I wanted to modify as little as possible.

Reply:
I guess that makes sense...for any functionality we're still relying upon ParaDIS. You've switched it to use VisIt functionality here. Can't we just use that now everywhere and always?

I guess I don't see the value in maintaining two equiv. code paths here...one that works everywhere but Windows and one that works everywhere (presumably since it is using functionality in VisIt).

Reply:
I can see your point. But it is a paradis lib file that I didn't want to change too much. The original author has other portions of the library that use the same ifdef  for special coding when paradis is built inside of VisIt.  I just followed that style.
IMHO the plugin should have been structured to build against an external paradis library, instead of sucking in the library's source in this fashion.  But then I would be porting the entire paradis source to Windows now, instead of only what is needed for the plugin :)
 
There are other problems with this reader. See #3562, #2048.